### PR TITLE
test: case insensitive email address test on passwordless flow

### PR DESCRIPTION
**Pull request by danielgent**
Works! No code changes. Looks like it's just the mgmt api users endpoint that needs addressing...

### CODE REVIEW

**Review by markusahlstrand — COMMENTED** [review]
The question is if we store it lowercase? Currently you could create two users, one with lowercase and one with uppercase which isn't great. Could you add a test for that?
I think it makes sense that we lowercase on account creation which I'm not sure if we're doing?

**Comment by danielgent** [comment]
> The question is if we store it lowercase? Currently you could create two users, one with lowercase and one with uppercase which isn't great. Could you add a test for that? I think it makes sense that we lowercase on account creation which I'm not sure if we're doing?

Ok. I'll add another test where I do a new sign up with an uppercase email... is that what you mean? login2 lowercases all the email input... and I think this is from back in the Auth0 days so I'll check what they do

File: test/integration/flows/code-flow.spec.ts [hunk 1]
```diff
@@ -612,6 +614,204 @@ describe("code-flow", () => {
...
+    const sub = parseJwt(accessToken!).sub;
+
+    // this means we have created the user
+    expect(tokenResponse.status).toBe(302);
+
+    // Now check in database we are storing in lower case
+
+    const newLowercaseUser = await env.data.users.get("tenantId", sub);
+
+    expect(newLowercaseUser!.email).toBe("john-doe@example.com");
```
Comment (danielgent) [inline test/integration/flows/code-flow.spec.ts:810]: :see_no_evil:  we could still be doing magic lowercasing in the adapter but the previous test is checking that we don't do a new signup

**Comment by danielgent** [comment]
Login2 always converts the entered email to lowercase so this is me locally *not* doing that to see what happens

image


`auth0.js` still sends up the email as uppercase in the API call

But I receive the email!

image

When I log in, the id token is giving me lowercase (which is what we're displaying here)

image


And in the dashboard the user is all lowercased

image


*So I would say this PR is correct and good-to-go* :checkered_flag:

**Review by markusahlstrand — APPROVED** [review]
Looks good! I think we need to make there there no way to create a user with an uppercase email, so think we need to patch the management api as well. Or.. we add some pre-create hook that ensures that they are lower case in a single place.

**Pull request merged**